internal/provider/string_data_source_test.go

```diff
@@ -0,0 +1,76 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccNameDataSource(t *testing.T) {
+	// A naked ref name has all but a .digest
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`data "oci_name" "test" { name = %q }`, "sample"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.oci_name.test", "id", "index.docker.io/library/sample:latest"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "registry", "index.docker.io"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "repo", "library/sample"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "context", "index.docker.io/library/sample"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "tag", "latest"),
+					resource.TestCheckNoResourceAttr("data.oci_name.test", "digest"),
+				),
+			},
+		},
+	})
+
+	// A tagged ref name has all but a .digest
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`data "oci_name" "test" { name = %q }`, "cgr.dev/chainguard/sample:foo"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.oci_name.test", "id", "cgr.dev/chainguard/sample:foo"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "registry", "cgr.dev"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "repo", "chainguard/sample"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "context", "cgr.dev/chainguard/sample"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "tag", "foo"),
+					resource.TestCheckNoResourceAttr("data.oci_name.test", "digest"),
+				),
+			},
+		},
+	})
+
+	// A digest ref name has everything (including a cursed tag)
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`data "oci_name" "test" { name = %q }`, "sample@sha256:1234567890123456789012345678901234567890123456789012345678901234"),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.oci_name.test", "id", "index.docker.io/library/sample@sha256:1234567890123456789012345678901234567890123456789012345678901234"),
+					resource.TestCheckResourceAttr("data.oci_name.test", "tag", "unused@sha256:1234567890123456789012345678901234567890123456789012345678901234"),
```

Comment:
I'd expect this to be the empty string

Reply:
I wanted `.tag` to always return something usable by the helm charts that only accept tags, something like below is faaaaar too common:

```
 image: {{ .Values.image.registry }}:${{ .Values.image.tag }}
```